includes/Modules/AdSense.php

```diff
@@ -731,25 +685,90 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 				case 'urlchannels':
 					// TODO: Parse this response to a regular array.
 					return $response->getItems();
-				case 'earning-today':
-				case 'earning-yesterday':
-				case 'earning-samedaylastweek':
-				case 'earning-7days':
-				case 'earning-prev7days':
-				case 'earning-this-month':
-				case 'earning-this-month-last-year':
-				case 'earning-28days':
-				case 'earning-prev28days':
-				case 'earning-daily-this-month':
-				case 'earnings-this-period':
-					// TODO: Parse this response to a regular array.
+				case 'earnings':
 					return $response;
 			}
 		}
 
 		return $response;
 	}
 
+	/**
+	 * Gets an array of dates for the given named date range.
+	 *
+	 * @param string $date_range Named date range.
+	 *                           E.g. 'last-28-days'.
+	 *
+	 * @return array|WP_Error Array of [startDate, endDate] or WP_Error if invalid named range.
+	 */
+	private function date_range_to_dates( $date_range ) {
+		switch ( $date_range ) {
+			case 'today':
+				return array(
+					date( 'Y-m-d', strtotime( 'today' ) ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
+				);
+			case 'yesterday':
+				return array(
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'same-day-last-week':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+				);
+			case '7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'prev-7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '14 days ago' ) ),
+					date( 'Y-m-d', strtotime( '8 days ago' ) ),
+				);
+			// Intentional fallthrough.
+			case 'daily-this-month':
+			case 'this-month':
+				return array(
+					date( 'Y-m-01' ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
+				);
+			case 'this-month-last-year':
+				$last_year          = intval( date( 'Y' ) ) - 1;
+				$last_date_of_month = date( 't', strtotime( $last_year . '-' . date( 'm' ) . '-01' ) );
+
+				return array(
+					date( $last_year . '-m-01' ),
+					date( $last_year . '-m-' . $last_date_of_month ),
+				);
+			case '28-days':
+				return array(
+					date( 'Y-m-d', strtotime( '28 days ago' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'prev-28-days':
+				return array(
+					date( 'Y-m-d', strtotime( '56 days ago' ) ),
+					date( 'Y-m-d', strtotime( '29 days ago' ) ),
+				);
+			// Intentional fallthrough.
+			case 'last-7-days':
+			case 'last-14-days':
+			case 'last-28-days':
+			case 'last-90-days':
+				preg_match( '/last-(?P<days>\d+)-days/', $date_range, $matches );
+
+				return array(
+					date( 'Y-m-d', strtotime( "{$matches['days']} days ago" ) ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
```

Comment:
I'd prefer if we didn't redo this here and instead had a `default` clause which would simply call `$this->parse_date_range( $date_range )` (see above).

Reply:
Updated, although not using `default` to preserve error behavior if invalid/unknown range passed. We could make an exception for any `last-*-days` though?